File: solutions/go/NumberOfSegmentsInAString.go
import "unicode"

func countSegments(s string) int {
	var n int
	sign := true

	for _, c := range s {
		if unicode.IsSpace(c) {
			sign = true
		} else if sign {
			n++
			sign = false
		}
	}

	return n
}

File: solutions/c/NumberOfSegmentsInAString.c
int
countSegments(char *s)
{
    int i, n;
    bool sign = true;

    for (n = i = 0; i < strlen(s); i++) {
        if (isspace(s[i])) {
            sign = true;
        } else if (sign) {
            n++;
            sign = false;
        }
    }

    return n;
}

